### PR TITLE
DBZ-5485 Support wait/nowait clause in mariadb

### PR DESCRIPTION
**Pull request by harveyyue**
Refer https://issues.redhat.com/browse/DBZ-5485

### CODE REVIEW

File: sql/mysql/Positive-Technologies/MySqlParser.g4 [hunk 1]
```diff
@@ -912,7 +913,7 @@ assignmentField
     ;
 
 lockClause
-    : FOR UPDATE | LOCK IN SHARE MODE
+    : FOR UPDATE lockOption? | LOCK IN SHARE MODE lockOption?
```
Comment (KvanTTT) [inline sql/mysql/Positive-Technologies/MySqlParser.g4:916]:
```suggestion
    : (FOR UPDATE | LOCK IN SHARE MODE) lockOption?
```

File: sql/mysql/Positive-Technologies/MySqlLexer.g4 [hunk 2]
```diff
@@ -138,6 +138,7 @@ LINEAR:                              'LINEAR';
 LINES:                               'LINES';
 LOAD:                                'LOAD';
 LOCK:                                'LOCK';
+LOCKED:                              'LOCKED';
```
Comment (KvanTTT) [inline sql/mysql/Positive-Technologies/MySqlLexer.g4:141]: Should it be added to [keywordsCanBeId](https://github.com/antlr/grammars-v4/blob/master/sql/mysql/Positive-Technologies/MySqlParser.g4#L2591)?

File: sql/mysql/Positive-Technologies/MySqlLexer.g4 [hunk 3]
```diff
@@ -189,6 +190,7 @@ SET:                                 'SET';
 SEPARATOR:                           'SEPARATOR';
 SHOW:                                'SHOW';
 SIGNAL:                              'SIGNAL';
+SKIP_:                               'SKIP';
```
Comment (KvanTTT) [inline sql/mysql/Positive-Technologies/MySqlLexer.g4:193]: Should it be added to [keywordsCanBeId](https://github.com/antlr/grammars-v4/blob/master/sql/mysql/Positive-Technologies/MySqlParser.g4#L2591)?

File: sql/mysql/Positive-Technologies/MySqlParser.g4 [hunk 4]
```diff
@@ -2656,7 +2667,7 @@ keywordsCanBeId
     | WORK | WRAPPER | X509 | XA | XA_RECOVER_ADMIN | XML
     // MariaDB
     | VIA | LASTVAL | NEXTVAL | SETVAL | PREVIOUS | PERSISTENT | REPLICATION_MASTER_ADMIN | REPLICA | READ_ONLY_ADMIN | FEDERATED_ADMIN | BINLOG_MONITOR | BINLOG_REPLAY
-    | ENCRYPTED | ENCRYPTION_KEY_ID
+    | ENCRYPTED | ENCRYPTION_KEY_ID | SKIP_ | LOCKED
```
Comment (harveyyue) [inline sql/mysql/Positive-Technologies/MySqlParser.g4:2670]: Yes, have added here.

File: sql/mysql/Positive-Technologies/MySqlParser.g4 (hunk 1 above)
Comment (harveyyue) [reply sql/mysql/Positive-Technologies/MySqlParser.g4]: fixed it.

**Pull request merged**